### PR TITLE
Fix status code for response of Post and Delete requests

### PR DESCRIPTION
**Pull request by ssshekhu53**
- Added types `PostResponder` and `DeleteResponder` implementing Responder interface to write responses for post and delete requests respectively
- Updated `ServeHTTP` method to create instances of `Responder` on the basis of request methods
- For details refer https://github.com/gofr-dev/gofr/issues/446

### CODE REVIEW

**Comment by srijan-27** [comment]
@ssshekhu53 - It is not advisable to create new duplicate responders just for the sake of adding statusCodes in response.
We would need to look into a better approach to resolve this.

**Comment by shashank-zopsmart** [comment]
> @ssshekhu53 - It is not advisable to create new duplicate responders just for the sake of adding statusCodes in response. We would need to look into a better approach to resolve this.

Have removed the new responders and added a field `method` in responder struct. Factory function for responder has been updated too to accept ResponseWriter and method in the parameter.  `HTTPStatusFromRequestMethod` will deduce the status code on the basis of http method for success cases.

File: pkg/gofr/http/responder.go [hunk 1]
```diff
@@ -65,6 +70,22 @@ func (r Responder) HTTPStatusFromError(err error) (status int, errObj interface{
 	}
 }
 
+// HTTPStatusFromRequestMethod maps request method to HTTP status codes.
+func (r Responder) HTTPStatusFromRequestMethod() int {
```
Comment (srijan-27) [inline pkg/gofr/http/responder.go:74]: This method is unnecessary addition. Could be included in `HTTPStatusFromErr` method itself.

**Review by srijan-27 — CHANGES_REQUESTED** [review]

File: pkg/gofr/handler_test.go [hunk 2]
```diff
@@ -24,17 +24,20 @@ var (
...
 		data       interface{}
 		err        error
 		statusCode int
 	}{
-		{"data is nil and error is nil", nil, nil, http.StatusOK},
-		{"data is mil, error is not nil", nil, errTest, http.StatusInternalServerError},
+		{"method is get, data is nil and error is nil", http.MethodGet, nil, nil, http.StatusOK},
+		{"method is get, data is mil, error is not nil", http.MethodGet, nil, errTest, http.StatusInternalServerError},
+		{"method is post, data is nil and error is nil", http.MethodPost, nil, nil, http.StatusCreated},
+		{"method is delete, data is nil and error is nil", http.MethodDelete, nil, nil, http.StatusNoContent},
```
Comment (srijan-27) [inline pkg/gofr/handler_test.go:35]: Since, data is nil for every case, so either we can include tests with data not being nil or we can remove the field and assume it `nil` by default.

**Review by srijan-27 — APPROVED** [review]

**Review by vipul-rawat — APPROVED** [review]
